pkg/prometheus/server/operator.go

```diff
@@ -354,15 +354,12 @@ func New(ctx context.Context, restConfig *rest.Config, c operator.Config, logger
 		}
 	}
 
-	endpointSliceSupported, err := k8sutil.IsAPIGroupVersionResourceSupported(o.kclient.Discovery(), schema.GroupVersion{Group: "discovery.k8s.io", Version: "v1"}, "endpointslices")
+	k8sAPIEndpointSliceSupported, err := k8sutil.IsAPIGroupVersionResourceSupported(o.kclient.Discovery(), schema.GroupVersion{Group: "discovery.k8s.io", Version: "v1"}, "endpointslices")
 	if err != nil {
 		level.Warn(o.logger).Log("msg", "failed to check if the API supports the endpointslice resources", "err ", err)
 	}
-	level.Info(o.logger).Log("msg", "Kubernetes API capabilities", "endpointslices", endpointSliceSupported)
-	// The operator doesn't yet support the endpointslices API.
-	// See https://github.com/prometheus-operator/prometheus-operator/issues/3862
-	// for details.
-	o.endpointSliceSupported = false
+	level.Info(o.logger).Log("msg", "Kubernetes API capabilities", "endpointslices", k8sAPIEndpointSliceSupported)
+	o.endpointSliceSupported = k8sAPIEndpointSliceSupported
```

Comment:
Same comments here as I've made above :)

Reply:
I have implemented this as in https://github.com/prometheus-operator/prometheus-operator/pull/6672#discussion_r1681835729